pkg/controller/memberstatus/memberstatus_controller.go

```diff
@@ -0,0 +1,305 @@
+
+	// kubefed not found
+	reasonHostConnectionNotFound = "KubefedNotFound"
+	errMsgHostConnectionNotFound = "the host connection was not found"
+)
+
+// statusComponentTags are used in the overall condition to point out which components are not ready
+type statusComponentTag string
+
+const (
+	memberOperator statusComponentTag = "memberOperator"
+	hostConnection statusComponentTag = "hostConnection"
+)
+
+// Add creates a new MemberStatus Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, _ *configuration.Config) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) *ReconcileMemberStatus {
+	return &ReconcileMemberStatus{
+		client:         mgr.GetClient(),
+		scheme:         mgr.GetScheme(),
+		getHostCluster: cluster.GetHostCluster,
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r *ReconcileMemberStatus) error {
+	// create a new controller
+	c, err := controller.New("memberstatus-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// watch for changes to primary resource MemberStatus
+	err = c.Watch(&source.Kind{Type: &toolchainv1alpha1.MemberStatus{}}, &handler.EnqueueRequestForObject{}, predicate.GenerationChangedPredicate{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileMemberStatus implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileMemberStatus{}
+
+// ReconcileMemberStatus reconciles a MemberStatus object
+type ReconcileMemberStatus struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client         client.Client
+	scheme         *runtime.Scheme
+	getHostCluster func() (*cluster.FedCluster, bool)
+}
+
+// Reconcile reads the state of toolchain member cluster components and updates the MemberStatus resource with information useful for observation or troubleshooting
+func (r *ReconcileMemberStatus) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling MemberStatus")
+
+	// fetch the MemberStatus
+	memberStatus := &toolchainv1alpha1.MemberStatus{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, memberStatus)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			return reconcile.Result{}, nil
+		}
+		reqLogger.Error(err, "Unable to fetch MemberStatus resource")
+		return reconcile.Result{}, err
+	}
+
+	err = r.aggregateAndUpdateStatus(reqLogger, memberStatus)
+	if err != nil {
+		reqLogger.Error(err, "Failed to update status")
+	}
```

Comment:
I thought about this but I was worried because I read that it will requeue immediately. BUT it looks like if I do `return reconcile.Result{RequeueAfter: defaultRequeueTime}, err` then it respects the  `RequeueAfter` time. I tested it out and it works.

Done in 6f77a0e80d789b3953b2e1269a1d34f1f1effd7d

Reply:
excellent solution :+1: :1st_place_medal: